verifiers/internal/gha/provenance.go

```diff
@@ -297,6 +299,45 @@ func VerifyProvenance(env *dsselib.Envelope, provenanceOpts *options.ProvenanceO
 		if err := isValidDelegatorBuilderID(prov); err != nil {
 			return err
 		}
+
+		id, err := prov.BuilderID()
```

Comment:
can we simply / replace this new code by:
```
if byob && builderOpts.ExpectedIDPath != nil{
   return verifyBuilderIDPath(prov, provenanceOpts.ExpectedBuilderIDPath) 
}
```

And implement `verifyBuilderIDPath` similarly to `verifyBuilderIDLooseMatch`

Reply:
Yes we can. Added this added this on new branch will close this pr and submit one with this refactor.